tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +55,1010 @@ def test_convert_labels_to_ids(domain: Domain):
+            {"action_name": [moodbot_features["actions"]["utter_did_that_help"]]},
+        ],
+        [
+            {},
+            {
+                "intent": [moodbot_features["intents"]["greet"]],
+                "action_name": [moodbot_features["actions"]["action_listen"]],
+            },
+            {"action_name": [moodbot_features["actions"]["utter_greet"]]},
+            {
+                "intent": [moodbot_features["intents"]["mood_unhappy"]],
+                "action_name": [moodbot_features["actions"]["action_listen"]],
+            },
+            {"action_name": [moodbot_features["actions"]["utter_cheer_up"]]},
+            {"action_name": [moodbot_features["actions"]["utter_did_that_help"]]},
+            {
+                "intent": [moodbot_features["intents"]["deny"]],
+                "action_name": [moodbot_features["actions"]["action_listen"]],
+            },
+        ],
+    ]
+    if max_history is not None:
+        expected_features = [x[-max_history:] for x in expected_features]
+
+    assert actual_features is not None
+    assert len(actual_features) == len(expected_features)
+
+    for actual, expected in zip(actual_features, expected_features):
+        assert compare_featurized_states(actual, expected)
+
+    expected_labels = np.array([[0, 15, 0, 12, 13, 0, 14]]).T
+
+    assert actual_labels is not None
+    assert actual_labels.shape == expected_labels.shape
+    assert np.all(actual_labels == expected_labels)
+
     # moodbot doesn't contain e2e entities
     assert not any([any(turn_tags) for turn_tags in entity_tags])
 
 
-def test_featurize_trackers_with_max_history_tracker_featurizer(moodbot_domain: Domain):
+@pytest.mark.parametrize("remove_duplicates", [True, False])
+@pytest.mark.parametrize("max_history", [None, 2])
+def test_deduplicate_featurize_trackers_with_max_history_tracker_featurizer(
+    moodbot_tracker: DialogueStateTracker,
+    moodbot_domain: Domain,
+    moodbot_features: Dict[Text, Dict[Text, Features]],
+    remove_duplicates: bool,
+    max_history: Optional[int],
+):
     state_featurizer = SingleStateFeaturizer()
-    tracker_featurizer = MaxHistoryTrackerFeaturizer(state_featurizer)
+    tracker_featurizer = MaxHistoryTrackerFeaturizer(
+        state_featurizer, max_history=max_history, remove_duplicates=remove_duplicates
+    )
 
-    tracker = tracker_from_dialogue_file(
-        "data/test_dialogues/moodbot.json", moodbot_domain
+    # Add Duplicate moodbot_tracker states should get removed.
+    actual_features, actual_labels, entity_tags = tracker_featurizer.featurize_trackers(
+        [moodbot_tracker, moodbot_tracker], moodbot_domain, RegexInterpreter()
+    )
+
+    expected_features = [
+        [{},],
+        [
+            {},
+            {
+                "intent": [moodbot_features["intents"]["greet"]],
+                "action_name": [moodbot_features["actions"]["action_listen"]],
```

Comment:
Please use the constants `INTENT`, `ACTION_NAME`, `ACTION_LISTEN` wherever possible.